pkg/reconcilers/base_reconciler.go

```diff
@@ -217,13 +217,17 @@ func (b *BaseReconciler) SetOwnerReference(owner, obj common.KubernetesObject) e
 
 // EnsureOwnerReference sets owner as a Controller OwnerReference on owned
 // returns boolean to notify when the object has been updated
+// Resource ownerReference isn't set if the resource is already owned by another controller
 func (b *BaseReconciler) EnsureOwnerReference(owner, obj common.KubernetesObject) (bool, error) {
 	changed := false
 
 	originalSize := len(obj.GetOwnerReferences())
-	err := b.SetOwnerReference(owner, obj)
-	if err != nil {
-		return false, err
+
+	if !isOwnedByAnotherController(obj) {
```

Comment:
The `controller-runtime`'s `SetControllerReference` method already checks that for you, returning `AlreadyOwnedError`. We can check for the errors, if it's already owned error, skip it, otherwise return error.

Reply:
Good suggestion - removed and refactored the error handling. Thanks